src/views/course_pairings.vue

```diff
@@ -0,0 +1,220 @@
+            }
+
+            let url = `{
+                "what":"course_pairings",
+                "user_id":"${this.$route.params.user_id}",
+                "term":"${term_where}",
+                "course_code":"${course_where}",
+                "column":"${column_name}"
+                }`;
+
+            url = JSON.parse(url);
+            fetch("get_info.php?" + generate_query_string(url))
+                .then(res => res.json())
+                .then(data => this.parseData(data))
+                .catch(err => {
+                    this.$emit("error", err.toString());
+                });
+        },
+
+        /**
+         * Function parses data from API call that will be rendered on the
+         * spreadsheet and select component.
+         */
+        parseData: function(data) {
+            var pairings = {};
+            for (let element of data.DATA) {
+                // Create an associative array with keys as course code and values
+                // as an array of instructors/tas for that couse.
+                pairings[element.course_code] =
+                    pairings[element.course_code] || [];
+                pairings[element.course_code].push({
+                    name: element.name,
+                    user_id: element.user_id
+                });
+
+                // Pushing list of distinct terms
+                if (this.list_terms.indexOf(element.term) < 0) {
+                    this.list_terms.push(element.term);
+                }
+            }
+            var user_associations = Object.keys(pairings).map(course => {
+                if (this.courses_list.indexOf(course) < 0) {
+                    this.courses_list.push(course);
+                }
+                return [course].concat(
+                    pairings[course].map(x => {
+                        return x;
+                    })
+                );
+            });
+            this.user_associations = user_associations;
+            this.createTable(user_associations);
+        },
+
+        /**
+         * Function creates spreadsheet component.
+         */
+        createTable: function(user_associations) {
+            var container = document.getElementById("table");
+
+            // Finding longest array
+            var longest = -1;
+            var index_longest = -1;
+            for (var index = 0; index < user_associations.length; index++) {
+                if (user_associations[index].length > longest) {
+                    longest = user_associations[index].length;
+                    index_longest = index;
+                }
+            }
+
+            //Creating column headers
+            var headers = [];
+            headers.push("Courses");
+            for (var index = 1; index < longest; index++) {
+                headers.push(this.column + " " + index);
+            }
+            this.tableSettings.colHeaders = headers;
+
+            // Creating table data
```

Comment:
What is example input and output to this? I think it can be done in a more compact way.